### PR TITLE
Keys.java, RequireTimeBefore.java Serialization/Deserialization Fix + AssetService Expansion + Updated Used By Section

### PR DESCRIPTION
**Pull request by edridudi**
Default Constructor for Deserialization.

### CODE REVIEW

File: src/main/java/com/bloxbean/cardano/client/crypto/Keys.java [hunk 1]
```diff
@@ -1,27 +1,14 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+public class Keys {
 
-    public void setSkey(SecretKey skey) {
-        this.skey = skey;
-    }
-}
+    private SecretKey skey;
+    private VerificationKey vkey;
```
[the remark this answers is not in the document]
Comment (edridudi) [reply src/main/java/com/bloxbean/cardano/client/crypto/Keys.java:13]: ignore

File: src/main/java/com/bloxbean/cardano/client/crypto/Keys.java [hunk 2]
```diff
@@ -1,27 +1,14 @@
...
 
-    public void setVkey(VerificationKey vkey) {
-        this.vkey = vkey;
-    }
-
-    public SecretKey getSkey() {
-        return skey;
-    }
+@Data
+@AllArgsConstructor
```
[the remark this answers is not in the document]
Comment (edridudi) [reply src/main/java/com/bloxbean/cardano/client/crypto/Keys.java:8]: ignore

File: src/main/java/com/bloxbean/cardano/client/crypto/Keys.java [hunk 3]
```diff
@@ -1,27 +1,14 @@
...
+import lombok.NoArgsConstructor;
 
-    public void setVkey(VerificationKey vkey) {
-        this.vkey = vkey;
-    }
-
-    public SecretKey getSkey() {
-        return skey;
-    }
+@Data
```
[the remark this answers is not in the document]
Comment (edridudi) [reply src/main/java/com/bloxbean/cardano/client/crypto/Keys.java:7]: ignore

File: src/main/java/com/bloxbean/cardano/client/backend/model/AssetAddress.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@AllArgsConstructor
+@ToString
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class AssetAddress {
+
+    /**
+     * Address containing the specific asset
+     */
+    private String address;
```
[the remark this answers is not in the document]
Comment (edridudi) [reply src/main/java/com/bloxbean/cardano/client/backend/model/AssetAddress.java:23]: ignore

File: src/main/java/com/bloxbean/cardano/client/backend/model/AssetAddress.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    /**
+     * Asset quantity on the specific address
+     */
+    private String quantity;
```
[the remark this answers is not in the document]
Comment (edridudi) [reply src/main/java/com/bloxbean/cardano/client/backend/model/AssetAddress.java:27]: ignore

File: src/main/java/com/bloxbean/cardano/client/backend/model/AssetAddress.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.*;
+
+/**
+ * AssetAddress
+ */
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
```
[the remark this answers is not in the document]
Comment (edridudi) [reply src/main/java/com/bloxbean/cardano/client/backend/model/AssetAddress.java:14]: ignore

**Comment by satran004** [comment]
@edridudi, Thanks a lot for the PR. All changes look good to me. I am merging this PR.

Some possible future changes. 
- Update RequireTimeAfter.java (similar to RequireTimeBefore)
- Enhance other composite script class (ScriptAll, ScriptAtleast..) to support JSON serialization / Deserialization
- Throw UnsupportedOperationException for unimplemented methods in GqlAssetService.java

**Pull request merged**